### ISSUE TITLE
Add README for each project in repo

### ISSUE THREAD

**Issue by captainsafia** [dotnet/aspnetcore#27561]
Part of https://github.com/dotnet/aspnetcore/issues/27380.

Let's improve the project-specific docs in the repo. For larger projects (Blazor, MVC, SignalR) the docs should contain information about the project's dependency, test setup, an overview of the folder structure, etc. For smaller projects, the doc should contain a description of the contents and what they might be useful for.

**Comment by blowdart** [comment]
I really wish you would keep the standard security text in the parent readme in each readme as well. If you're overriding all of this it's important to highlight the right way to report security bugs.

**Comment by JamesNK** [comment]
Rather than copy parts of the main readme, we should add a link to it. e.g.

```
## More Information

For more information, see the [ASP.NET Core README](../README.md).
```

**Comment by captainsafia** [comment]
@JamesNK Good idea. I've updated the template in the issue with this.

**Comment by dougbu** [comment]
> SiteExtensions (@dougbu)

@jkotalik I have mucked in that area a few times but don't really get how the extensions are _used_. Could you take that README too or find another ~~victim~~ delegate❔

**Comment by jkotalik** [comment]
I can try.

**Comment by captainsafia** [comment]
Hi folks! Adjusted the template based on some feedback from Pranav. To avoid repeatedly stating that users can run the `build.cmd` command in the directory, we'll instead refer to the "build subset of code" section in the top-level docs so we have less duplication there. We should still include any project-specific build instructions like helpful flags to pass to the build command or dependencies that need to be installed.

Ditto for the test section.

**Comment by halter73** [comment]
I don't think we should worry about duplicating small inline sample commands like those in https://github.com/dotnet/aspnetcore/blob/45d77afa717e26576c28ee97d3638ce25532d0ed/src/SignalR/README.md. Having to click links to see what the one-line build and test commands are is annoying. Linking to the top-level docs is great, but deduping stuff like one-line build and test commands should be a non-goal in my opinion.

**Comment by Rick-Anderson** [comment]
Feel free to mention me for an edit review.
